utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/collection.scala

```diff
@@ -0,0 +1,135 @@
+package pl.touk.nussknacker.engine.util.functions
+
+import pl.touk.nussknacker.engine.api.generics.GenericType
+import pl.touk.nussknacker.engine.api.{Documentation, HideToString, ParamName}
+
+import java.util.{Collections, Objects}
+import scala.collection.immutable.ListMap
+import scala.jdk.CollectionConverters.{CollectionHasAsScala, _}
+
+object collection extends HideToString {
+
+  @Documentation(description = "Concatenates two lists")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListAdditionTyping])
+  def concat[T](@ParamName("list1") list1: java.util.List[T], @ParamName("list2") list2: java.util.List[T]): java.util.List[T] = (list1.asScala.toList ++ list2.asScala).asJava
+
+  @Documentation(description = "Merges two maps. Values in the first map will be overwritten with values from the second map if keys are the same")
+  @GenericType(typingFunction = classOf[TypingFunctions.MapMergeTyping])
+  def merge[K, V](@ParamName("map1") map1: java.util.Map[K, V], @ParamName("map2") map2: java.util.Map[K, V]): java.util.Map[K, V] = {
+    val merged = new java.util.LinkedHashMap[K, V](map1)
+    merged.putAll(map2)
+    merged
+  }
+
+  @Documentation(description = "Returns the smallest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListElementTyping])
+  def min[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = {
+    checkIfComparable(list)
+    Collections.min[T](list)
+  }
+
+  @Documentation(description = "Returns the largest element (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListElementTyping])
+  def max[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): T = {
+    checkIfComparable(list)
+    Collections.max[T](list)
+  }
+
+  @Documentation(description = "Returns a slice of the list starting with start index (inclusive) and ending at stop index (exclusive)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListTyping])
+  def slice[T](@ParamName("list") list: java.util.Collection[T], @ParamName("start") start: Int, @ParamName("stop") stop: Int): java.util.List[T] = {
+    list.asScala.slice(start, stop).toList.asJava
+  }
+
+  @Documentation(description = "Returns a sum of all elements")
+  def sum[T <: java.lang.Number](@ParamName("listOfNumbers") list: java.util.Collection[T]): java.lang.Number = {
+    list.asScala.map(x => new java.math.BigDecimal(x.toString)).reduce((a, b) => a.add(b))
+  }
+
+  @Documentation(description = "Returns a list of all elements sorted in ascending order (elements must be comparable)")
+  @GenericType(typingFunction = classOf[TypingFunctions.ListTyping])
+  def sorted[T<:Comparable[T]](@ParamName("list") list: java.util.Collection[T]): java.util.List[T] = {
```

Comment:
WDYT aboud adding `sortedAsc` and `sortedDesc` variants like for maps?

Reply:
Good idea :+1: